packages/api/src/controllers/webhook.test.ts

```diff
@@ -288,8 +290,22 @@ describe("controllers/webhook", () => {
       });
       expect(setActiveRes).toBeDefined();
       expect(setActiveRes.status).toBe(204);
-      // const setActiveResJson = await setActiveRes.json()
-      // expect(setActiveResJson).toBeDefined()
+
+      let hooksCalled = 0;
+      for (let i = 0; i < 5 && hooksCalled < 2; i++) {
+        await sleep(500);
+
+        hooksCalled = 0;
+        for (const id of [webhookResJson.id, webhookResJson2.id]) {
+          const res = await client.get(`/webhook/${id}`);
+          expect(res.status).toBe(200);
+          const { status } = (await res.json()) as Webhook;
+          if (status.lastTriggeredAt >= now) {
+            hooksCalled++;
+          }
+        }
+      }
+      expect(hooksCalled).toBe(2);
```

Comment:
FTR: this was the very cryptic failure I was getting from the tests. What has happening was that, after creating the separate jobs DB pool, the test below (`records webhook logs`) started failing because the webhook created _there_ was begin called twice, one additional time with the event triggered in _this_ test.

The problem was that this test never really waited (or even checked) that the webhooks created here were actually triggered. So the test finished while leaving the background runner calling the webhooks for it. This probably caused some flakiness every now and then as well.

What has happening was that, after switching to the separate DB pool, the very first event handled by the `WebhookCannon` took a little longer since it had to setup a new connection with Postgres (that hadn't been already initialized from the usual test setup). This small delay was already enough for the next test to start and create another webhook, which then got called for the same event. I could reproduce this by making a dummy query on the `jobsDb` in this test, which then caused the failure to stop cause the pool was then already initialized.

Fixed this by correcting this test logic tho, which was not even checking that the webhooks were being called. Explaining here only cause it seems so unrelated to the changes here, but they actually were.